bl.js

```diff
@@ -49,16 +47,21 @@ BufferList.prototype._offset = function (offset) {
 }
 
 BufferList.prototype.append = function (buf) {
-  var isBuffer = Buffer.isBuffer(buf) ||
-                 buf instanceof BufferList
+  if (buf instanceof BufferList) {
+    // unwrap argument into individual BufferLists
+    for (var i = 0; i < buf._bufs.length; i++)
+      this.append(buf._bufs[i])
+    return
+  }
 
   // coerce number arguments to strings, since Buffer(number) does
   // uninitialized memory allocation
   if (typeof buf == 'number')
     buf = buf.toString()
 
-  this._bufs.push(isBuffer ? buf : new Buffer(buf))
-  this.length += buf.length
```

Comment:
pretty sure there was a bug in here if you pass in a string with some unicode multibyte, in fact we don't really do any unicode testing so who knows what's lurking in here